processor/spanmetricsprocessor/processor.go

```diff
@@ -60,9 +63,14 @@ type exemplarData struct {
 }
 
 type metricKey string
+type resourceKey string
 
 type processorImp struct {
-	lock   sync.RWMutex
+	// Based on OTEL Collector design, each pipeline will always get its own instance of the processor. ConsumeTraces()
+	// of each instance should never be called concurrently in practice.
+	// This processor is stateful. Due to the nature of its logic, the concurrent executions of ConsumeTraces() will
+	// output incorrect data. This lock forces the ConsumeTraces() can only execute in serial.
+	lock   sync.Mutex
```

Comment:
This lock is not used to protect any variable. As the comment says, the processor's instance should never be called concurrently. Otherwise, the aggregated data will be incorrect.

See the discussion here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6717#discussion_r779312693.

It is worth noting that most of the methods of `ProcessImp` are not concurrent safe. We only guarantee this  `ConsumeTraces()` is concurrent safe. But it actually can on be executed in serial to make sure the aggregation logic is correct.